pennylane/ops/functions/equal.py

```diff
@@ -305,25 +304,44 @@ def _equal_hamiltonian(op1: Hamiltonian, op2: Observable, **kwargs):
 
 @_equal.register
 # pylint: disable=unused-argument
-def _equal_measurements(op1: MeasurementProcess, op2: MeasurementProcess, **kwargs):
+def _equal_measurements(
+    op1: MeasurementProcess,
+    op2: MeasurementProcess,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
     """Determine whether two MeasurementProcess objects are equal"""
+    if op1.wires != op2.wires:
+        return False
```

Comment:
this was true before so this is not relevant for the release, but I am curious what we consider "equal". ie. in the following, should `a` and `b` be considered equal?

```pycon
>>> z0 = qml.PauliZ(0)
>>> x1 = qml.PauliX(1)
>>> a = qml.expval(qml.prod(z0, x1))
>>> b = qml.expval(qml.prod(x1, z0))
>>> qml.math.allequal(a.obs.matrix(), b.obs.matrix(wire_order=[0, 1]))
True
>>> a.wires == b.wires
False
```

I guess this is strongly tied to the definition of `expval` as highlighted by my favourite bug this release :)

Reply:
Actually a pretty good point.

We already have `qml.equal(qml.prod(z0, x1), qml.prod(x1, z0))` as True, so we should also have the expectation values of the two be equal.  Easily able to accommodate by switching the order of checks. I'm adding this case as well.

Reply:
sounds good, but now I'm wondering if this is considered breaking because it used to return `False` but would now return `True`? I didn't realize the bases are considered equal, but it's weird that it would change just because it's wrapped in a MP so the change certainly sounds sensible. I'm ok with whatever you prefer to do for this release (if someone's work is dependent on `expval(z0 @ x1) != expval(x1 @ z0)`... idk what to say)

Reply:
I think this falls under "aligning actual behaviour with expected results" instead of "changing what is expected".